src/restic/backend/gs/config.go

```diff
@@ -0,0 +1,71 @@
+package gs
+
+import (
+	"path"
+	"strings"
+
+	"restic/errors"
+	"restic/options"
+)
+
+// Config contains all configuration necessary to connect to an gcs compatible
+// server.
+type Config struct {
+	Location    string
+	ProjectID   string
+	JSONKeyPath string
+	Bucket      string
+	Prefix      string
+	Layout      string `option:"layout" help:"use this backend layout (default: auto-detect)"`
+
+	Connections uint `option:"connections" help:"set a limit for the number of concurrent connections (default: 20)"`
+}
+
+// NewConfig returns a new Config with the default values filled in.
+func NewConfig() Config {
+	return Config{
+		Connections: 20,
+	}
+}
+
+func init() {
+	options.Register("gs", Config{})
+}
+
+const defaultPrefix = "restic"
+
+// ParseConfig parses the string s and extracts the gcs config. The two
+// supported configuration formats are gs://host/bucketName/prefix and
+// gs:host:bucketName/prefix. The host can also be a valid gs region
+// name. If no prefix is given the prefix "restic" will be used.
+func ParseConfig(s string) (interface{}, error) {
+	switch {
+	case strings.HasPrefix(s, "gs://"):
+		s = s[5:]
+	case strings.HasPrefix(s, "gs:"):
+		s = s[3:]
+	default:
+		return nil, errors.New("gs: invalid format")
+	}
+	// use the first entry of the path as the endpoint and the
+	// remainder as bucket name and prefix
+	path := strings.SplitN(s, "/", 3)
```

Comment:
Is the endpoint relevant for other commands besides `init`?

I'd suggest using a format of `gcs:ENDPOINT:PATH`, so that it is consistent with the `s3` backend.